api/levity.proto

```diff
@@ -0,0 +1,91 @@
+syntax = "proto3";
+package levity;
+
+import "google/protobuf/empty.proto";
+
+option go_package = "github.com/tcsc/levity/api";
+
+
+service TaskManager {
+    // StartTask attempts to launch a task on the server, returning a handle
+    // to interact with the task via the other API end points. he launched
+    // process will run under, and inherit the permissions of, the account 
+    // of the user running the API server. No shell variable substitution
+    // will be performed.
+    rpc StartTask(StartTaskRequest) returns (StartTaskResponse) {}
+
+    // QueryTask fetches the current state of the task, e.g. Running, 
+    // Finished, etc.
+    rpc QueryTask(QueryTaskRequest) returns (QueryTaskResponse) {}
+
+    // SignalTasks requests that the task exit. The process will be given a
+    // SIGINT, and if the process does not exit after an arbitrary timeout it
+    // will be killed. Note that this call will NOT wait on the task to
+    // finish; the caller will need to poll with QueryTask to determine when
+    // the task has finished.
+    rpc SignalTask(SignalTaskRequest) returns (google.protobuf.Empty) {}
+
+    // FetchLogs returns the data written to stdout and stderr by the task. The
+    // log data from each stream is treated as an opaque series of bytes  
+    rpc FetchLogs(FetchLogsRequest) returns (FetchLogsResponse) {}
+
+    // Obvious extension is streaming the logs as they come in
+    // using gRPC streaming.
+}
+
+// TaskHandle stores an idetifier that uniquely identifies a task while it is
+// registered with the API server. IDs may be recycled during the lifetime of
+// the server process.
+message TaskHandle {
+    string id = 1;
+}
+
+message StartTaskRequest {
+    string binary = 1;
+    repeated string args = 2;
+    optional string workingDir = 3;
```

Comment:
Sorry to nitpick to the code that will basically be converted to auto-generated code, but case here is again inconsistent with the rest of the fields that use underscores.

Reply:
Yeah, fair enough. The ProtoBuf style guide says snake case, so may as well fix it now.

Reply:
Done